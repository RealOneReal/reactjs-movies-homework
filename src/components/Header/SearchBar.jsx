import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Box, TextField, IconButton } from "@mui/material";
import SearchIcon from "@mui/icons-material/Search";
import { inputSearch } from "../../redux/search/searchSlice";
import {
  fetchMoviesBySearch,
  fetchGenresConfig,
} from "../../redux/movies/moviesSlice";

const SearchBar = () => {
  const [value, setValue] = useState("");
  const dispatch = useDispatch();
  const { language } = useSelector((state) => state.search);

  const handleInput = ({ target }) => {
    setValue(target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    event.key === "Enter " && dispatch(inputSearch(value));
    dispatch(inputSearch(value));
    value && dispatch(fetchGenresConfig({ language }));
    value && dispatch(fetchMoviesBySearch({ value, pageNumber: 1, language }));
  };

  const styles = {
    formGroup: {
      width: { md: "400px", xs: "200px" },
      display: "flex",
      flexDirection: "row",
      columnGap: "8px",
    },
    textField: {
      width: {
        xs: "150px",
        md: "350px",
      },
      "& .MuiInputBase-input": {
        color: "primary.dark",
      },
    },
    iconButton: { width: "20px", color: "primary.dark" },
  };

  return (
    <Box sx={styles.formGroup} component="form" onSubmit={handleSubmit}>
      <TextField
        value={value}
        onChange={handleInput}
        type="text"
        placeholder="Movies,person,movie theaters"
        variant="standard"
        sx={styles.textField}
      />
      <IconButton sx={styles.iconButton} type="submit">
        <SearchIcon />
      </IconButton>
    </Box>
  );
};

export default SearchBar;
