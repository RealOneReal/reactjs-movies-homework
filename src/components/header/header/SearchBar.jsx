import { TextField } from "@mui/material";
import { useState } from "react";

const SearchBar = () => {
  const [value, setValue] = useState("");

  const handleInput = (event) => {
    setValue(event.target.value);
  };

  const styles = {
    textField: {
      width: "400px",
    },
  };

  return (
    <TextField
      value={value}
      onChange={handleInput}
      type="search"
      placeholder="Movies,person,movie theaters"
      variant="standard"
      sx={styles.textField}
    />
  );
};

export default SearchBar;
