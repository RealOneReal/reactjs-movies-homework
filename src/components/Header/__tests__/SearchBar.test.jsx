import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import SearchBar from "../SearchBar";

describe("SearchBar", () => {
  it("should be render", () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should change value", () => {
    const { getByRole } = render(<SearchBar />);
    const input = getByRole("textbox");
    userEvent.type(input, "hello from Brest");
    expect(input).toHaveValue("hello from Brest");
  });
});
