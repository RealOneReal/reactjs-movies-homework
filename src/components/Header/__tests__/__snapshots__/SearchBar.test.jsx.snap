// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should be render 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-rc5ywp-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Movies,person,movie theaters"
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
